src/game_logic/interfaces.ts

```diff
@@ -80,76 +81,58 @@ export interface Contract {
   readonly name: string
   // Long form description / flavor text
   readonly description: string
-  /**
-   * Look at the current game state and determine if this contract has been fulfilled
-   *
-   * changes `isCompeleted()` if has been completed
-   *
-   * DO NOT modify the game state
-   * @param state
-   */
-  check(state: GameState): boolean
-
-  /**
-   * Modify the current game state to apply the rewards of this contract
-   * @param state
-   */
-  reward(state: GameState): void
-
-  /**
-   * Indicates if this contract has been resolved
-   */
+  // Indicates if this contract has been resolved
   readonly completed: boolean
 }
 
 export interface MapState {
-  tiles: Array<Tile>,
-  size: {
-    x: number,
-    y: number
+  readonly tiles: ReadonlyArray<Tile>,
+  readonly size: {
+    readonly x: number,
+    readonly y: number
   }
 }
 
 export interface PlayerState {
-  resources: {
-    money: {
-      balance: number,
-      revenue: number
+  readonly resources: {
+    readonly money: {
+      readonly balance: number,
+      readonly revenue: number
     },
-    workers: {
-      max: number,
-      free: number
+    readonly workers: {
+      readonly max: number,
+      readonly free: number
     }
   },
-  victory: {
-    happiness: number,
-    goal: number
+  readonly victory: {
+    readonly happiness: number,
+    readonly goal: number
   },
-  contracts: {
-    completed: Array<Contract>,
-    open: Array<Contract>
+  readonly contracts: {
+    readonly completed: ReadonlyArray<Contract>,
+    readonly open: ReadonlyArray<Contract>
   }
 }
 
 /**
  * The state of the game, design to be transformed only through specific functions, but observed by any elements.
  */
 export interface GameState {
-  game: {
-    turnCounter: number
+  readonly game: {
+    readonly turnCounter: number
   },
-  player: PlayerState,
+  readonly player: PlayerState,
   // A flat-buffer of
-  map: MapState,
-  tileCatalog: TileCatalog
+  readonly map: MapState,
+  readonly tileCatalog: TileCatalog
```

Comment:
is this property of `GameState` used anywhere? you can add a `TODO` comment to consider removing it - it seems weird to me that the catalog belongs in this interface.

Reply:
You're absolutely right. I was planning on removing this, but my `// TODO` got lost in the stash.

Reply:
And the fact that nothing is erroring despite this line being here and nothing assigning to it suggests `type` is better than `interface`, generally...

Reply:
Oops, this has nothing to do with `interface` vs `type`. The real answer is that excess property checking only works when using a fresh object literal. Pushing a fix.